oxidation/parsec_cli/parsec_sequestre_dump/src/model.rs

```diff
@@ -0,0 +1,519 @@
+                    id: *id,
+                    version,
+                    created: *created.as_ref(),
+                    updated: *updated.as_ref(),
+                    size,
+                    children: vec![],
+                    content,
+                }
+            }
+            Manifest::Folder(FolderManifest {
+                author,
+                timestamp,
+                id,
+                version,
+                created,
+                updated,
+                children,
+                ..
+            })
+            | Manifest::Workspace(WorkspaceManifest {
+                author,
+                timestamp,
+                id,
+                version,
+                created,
+                updated,
+                children,
+                ..
+            }) => {
+                let children = children
+                    .into_iter()
+                    .map(|(name, id)| {
+                        vlob_atom::table
+                            .select(vlob_atom::blob)
+                            .filter(vlob_atom::vlob_id.eq((*id).as_ref()))
+                            .order_by(vlob_atom::timestamp.desc())
+                            .first::<Vec<u8>>(conn)
+                            .map(|x| Self::load(conn, key, name.as_ref().to_string(), &x))
+                    })
+                    .collect::<Result<Result<Vec<_>, _>, _>>()??;
+
+                let size = children.iter().map(|x| x.size).sum();
+
+                Self {
+                    name,
+                    author,
+                    timestamp: *timestamp.as_ref(),
+                    id: *id,
+                    version,
+                    created: *created.as_ref(),
+                    updated: *updated.as_ref(),
+                    size,
+                    children,
+                    content: String::new(),
+                }
+            }
+            _ => unreachable!(),
+        })
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct Workspace(pub Data);
+
+impl Workspace {
+    pub fn dump(&self, path: &str) -> Result<(), Box<dyn Error>> {
+        let data = serde_json::to_string_pretty(self)?;
+        std::fs::write(path, &data)?;
+
+        Ok(())
+    }
+
+    pub fn query_all(conn: &mut SqliteConnection, key: &SecretKey) -> Result<Self, Box<dyn Error>> {
+        let workspace_id = info::table.select(info::realm_id).first::<Vec<u8>>(conn)?;
+
+        let workspace = vlob_atom::table
+            .select(vlob_atom::blob)
+            .filter(vlob_atom::vlob_id.eq(workspace_id))
+            .first::<Vec<u8>>(conn)?;
+
+        Ok(Self(Data::load(conn, key, String::new(), &workspace)?))
+    }
+
+    #[cfg(test)]
+    pub fn load(path: &str) -> Self {
+        let data = std::fs::read(path).unwrap();
+        serde_json::from_slice(&data).unwrap()
+    }
+
+    #[cfg(test)]
+    pub fn init_db(conn: &mut SqliteConnection, device: &Device) {
```

Comment:
This is totally related to the test, you should have all this in a `tests` folder instead